x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1136,6 +1136,21 @@ private LogicalPlan doRule(LogicalPlan plan) {
                 }
                 plan = plan.transformExpressionsOnly(UnresolvedAttribute.class, ua -> resolveAttribute(ua, resolved));
             }
+            if (plan instanceof InlineStats stats) {
+                System.err.println("ADSFADFSAF " + stats);
+                // If the union-types resolution occurred in a child of the aggregate, we need to check the groupings
+                plan = stats.transformExpressionsOnly(FieldAttribute.class, UnionTypesCleanup::checkUnresolved);
+
+                // Aggregates where the grouping key comes from a union-type field need to be resolved against the grouping key
+                Map<Attribute, Expression> resolved = new HashMap<>();
+                for (Expression e : stats.groupings()) {
+                    Attribute attr = Expressions.attribute(e);
+                    if (attr != null && attr.resolved()) {
+                        resolved.put(attr, e);
+                    }
+                }
+                plan = plan.transformExpressionsOnly(UnresolvedAttribute.class, ua -> resolveAttribute(ua, resolved));
+            }
```

Comment:
This is an attempt to solve a problem with unit types that's come up. It's a copy of above, which ain't great. Consider this a merge blocker.

Reply:
The special casing for STATS was made obsolete in https://github.com/elastic/elasticsearch/pull/111475. I'd hope that it's also not required for INLINESTATS.